lib/Doctrine/Persistence/Mapping/Driver/ColocatedMappingDriver.php

```diff
@@ -0,0 +1,209 @@
+<?php
+
+namespace Doctrine\Persistence\Mapping\Driver;
+
+use Doctrine\Persistence\Mapping\MappingException;
+use FilesystemIterator;
+use RecursiveDirectoryIterator;
+use RecursiveIteratorIterator;
+use RecursiveRegexIterator;
+use ReflectionClass;
+use RegexIterator;
+
+use function array_merge;
+use function array_unique;
+use function assert;
+use function get_declared_classes;
+use function in_array;
+use function is_dir;
+use function preg_match;
+use function preg_quote;
+use function realpath;
+use function str_replace;
+use function strpos;
+
+/**
+ * The ColocatedMappingDriver reads the mapping metadata located near the code.
+ */
+abstract class ColocatedMappingDriver implements MappingDriver
```

Comment:
I wonder if we should make this a trait instead.

`ColocatedMappingDriver` as a type does not make sense; We'll probably never see `ColocatedMappingDriver` in a return type or `instanceof` check for instance. And all this class contains is just some shared implementation.

Reply:
Good point, I hadn't considered that. The fact that that class is hard to name is probably a hint at the fact that there should not be a type.